core/src/test/java/org/apache/iceberg/TestTableMetadata.java

```diff
@@ -35,6 +35,7 @@
 import org.apache.iceberg.TableMetadata.MetadataLogEntry;
```

Comment:
Does this include tests for the serde to/from json?

Reply:
Yep.

Reply:
This also tests backward compatibility.